File: daq/slc/src/dqm/src/Axis.cc
#include "Axis.hh"

#include <sstream>

using namespace B2DQM;

Axis::Axis()
  : _nbins(1), _min(0), _max(1), _title(""),
    _fixed_min(false), _fixed_max(false) {}

Axis::Axis(int nbin, double min, double max,
           const std::string& title)
  : _nbins(nbin), _min(min), _max(max), _title(title),
    _fixed_min(false), _fixed_max(false) {}

Axis::Axis(int nbin, double min, double max)
  : _nbins(nbin), _min(min), _max(max),
    _title(""), _fixed_min(false), _fixed_max(false) {}

Axis::Axis(const Axis& axis)
  : _nbins(axis._nbins), _min(axis._min), _max(axis._max),
    _title(axis._title), _fixed_min(false), _fixed_max(false) {}

Axis::~Axis() throw() {}

void Axis::set(int nbin, double min, double max)
{
  setNbins(nbin);
  setMin(min);
  setMax(max);
}

void Axis::set(int nbin, double min, double max, const std::string& title)
{
  set(nbin, min, max);
  setTitle(title);
}

void Axis::readObject(B2DAQ::Reader& reader) throw(B2DAQ::IOException)
{
  _title = reader.readString();
  setNbins(reader.readInt());
  setMin(reader.readDouble());
  setMax(reader.readDouble());
}

void Axis::writeObject(B2DAQ::Writer& writer) const throw(B2DAQ::IOException)
{
  writer.writeString(_title);
  writer.writeInt(getNbins());
  writer.writeDouble(getMin());
  writer.writeDouble(getMax());
}

const Axis& Axis::operator=(const Axis& axis)
{
  _nbins = axis._nbins;
  _min = axis._min;
  _max = axis._max;
  _title = axis._title;
  return *this;
}

File: daq/slc/extra/javalib/b2daq/src/b2daq/dqm/core/Axis.java
package b2daq.dqm.core;

public class Axis {

    private String name = "";
    private int nbins = 0;
    private double min = 0;
    private double max = 1;
    private boolean fixedMin = false;
    private boolean fixedMax = false;
    private String title = null;

    public Axis(int nbin, double min, double max, String label) {
        nbins = nbin;
        this.min = min;
        this.max = max;
        title = label;
    }

    public Axis() {
        this(1, 0, 1, "");
    }

    public Axis(int nbin, double min, double max) {
        this(nbin, min, max, "");
    }

    public Axis(Axis axis) {
        this(axis.nbins, axis.min, axis.max, axis.title);
    }

    public void copy(Axis axis) {
        nbins = axis.nbins;
        min = axis.min;
        max = axis.max;
        title = axis.title;
        fixedMin = axis.fixedMin;
        fixedMax = axis.fixedMax;
    }

    public void copyRange(Axis axis) {
        if (!fixedMin) {
            min = axis.min;
        }
        if (!fixedMax) {
            max = axis.max;
        }
    }

    public int getNbins() {
        return nbins;
    }

    public double getMax() {
        return max;
    }

    public double getMin() {
        return min;
    }

    public String getTitle() {
        return title;
    }

    public boolean isFixedMin() {
        return fixedMin;
    }

    public boolean isFixedMax() {
        return fixedMax;
    }

    public void setRange(int nbin, double min, double max) {
        nbins = nbin;
        this.min = min;
        this.max = max;
    }

    public void setNbins(int nbins) {
        this.nbins = nbins;
    }

    public void setMax(double max) {
        //if (!fixedMax) {
            this.max = max;
        //}
    }

    public void setMin(double min) {
        //if (!fixedMin) {
            this.min = min;
        //}
    }

    public void setTitle(String label) {
        title = label;
    }

    public void setFixedMin(boolean fix) {
        fixedMin = fix;
    }

    public void setFixedMax(boolean fix) {
        fixedMax = fix;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public String toString() {
        String script = "{"
                + "'label':'" + getTitle() + "',"
                + "'nbins':" + getNbins() + ","
                + "'min':" + getMin() + ","
                + "'max':" + getMax() + "}";
        return script;
    }

    public String getXML(String name) {
        String script = "<" + name + " "
                + "label=\"" + getTitle() + "\" "
                + "nbins=\"" + getNbins() + "\" "
                + "min=\"" + getMin() + "\" "
                + "max=\"" + getMax() + "\" ";
        if (isFixedMin()) {
            script += "fixed-min=\"" + isFixedMin() + "\" ";
        }
        if (isFixedMax()) {
            script += "fixed-max=\"" + isFixedMax() + "\" ";
        }
        script += "/>\n";
        return script;
    }

}
